### ISSUE TITLE
Questions about code usage

### ISSUE THREAD

**Issue by jcgriffis** [TingsterX/SpatialGradientParcellation#1]
Hello, 

I have a few questions about applying this code to my data.

The documentation says the usage is: 

./x_gradient.sh PathToFile/<lh.functional_similarity.nii.gz> <ThePath/lh.mask.nii.gz> <L.midthickness.surf.gii> <OutDirectory>

What is the procedure for creating the <lh.functional_similarity.nii.gz> file, and what is the file <lh.mask.nii.gz> supposed to contain?

In the x_gradient.sh script, the variable <fsaverage> is set to fsaverage5. What is this for? If my data are on the 10k_fs_lr surface and aligned to the Yerkes19 template, would I simply set this to 10k_fs_lr? Also, what should <FWHM1> and <FWHM2> be set to? 

If my data are already in gifti/cifti format, do I need to remove the first two calls to wb_command -metric-convert?

Basically, the data that I am working with were pre-processed by someone else and so I do not have access to the raw data files, but what I do have are surface timeseries files (dtseries.nii) and surface files (surf.gii) at 10k vertex resolution registered to the Yerkes19 template for each animal. I'm just trying to make sure I am applying this appropriately to these data.  

Thank you for your help!

**Comment by TingsterX** [comment]
Hi Joseph,

What is the procedure for creating the <lh.functional_similarity.nii.gz> file, and what is the file <lh.mask.nii.gz> supposed to contain?
- I didn't include the code for calculating the functional similarity here. The similarity of functional connectivity can be done using the matlab function 'paircorr_mod.m' or wb_command command '-cifti-correlation'. Note here, consistent with Wig2013, Gordon2014, the cortical feature is the similarity of functional connectivity, we can also use other cortical features as well (e.g. myelin, Glasser2016). 
- mask file contains 1/0. Any vertices=0 won't be included in the smooth and gradient calculation. If the FOV of your data covers the whole brain, you can use the standard mask *atlasroi* which only excludes the medial wall.

In the x_gradient.sh script, the variable is set to fsaverage5. What is this for? If my data are on the 10k_fs_lr surface and aligned to the Yerkes19 template, would I simply set this to 10k_fs_lr? Also, what should and be set to?

- This is the surface used to calculate the gradient on. The surface I used in Xu2016 paper was fsaverage5 so I left the fsaverage5 there. In your case, yes, it is 10k_fs_LR surface. Of note, it's important to use the individual midthickness surface. So the shape of the surface used here matches with the individual anatomical morphology but the vertices can have a one-to-one correspondence to template surface (e.g. Yerkes19 10k surface for macaque data). 

If my data are already in gifti/cifti format, do I need to remove the first two calls to wb_command -metric-convert?

- If your data is in gifti format, then no need to do -metric-convert. I used nifti format because it's faster to load nifti rather than gifti in matlab in the earlier calculation steps. For cifti format, Workbench actually has '-cifti-gradient'. In that case, you'll just need to apply your mask when you create the cifti data.

Basically, the data that I am working with were pre-processed by someone else and so I do not have access to the raw data files, but what I do have are surface timeseries files (dtseries.nii) and surface files (surf.gii) at 10k vertex resolution registered to the Yerkes19 template for each animal. I'm just trying to make sure I am applying this appropriately to these data.

- Just make sure that the surface matches with your volume data where it was projected from. Take HCP data as an example, if we use the HCP preprocessed data, this would be subFolder/MNINonLinear/fsaverage_32kLR/subID.Hemi.midthickness.32k_fs_LR.surf.gii. Of note, if by any chance, your surafce data is aligned to native T1 and project to the surface from there, then we can use subFolder/T1w/fsaverage_LR32k/subID.Hemi.midthickness.32k_fs_LR.surf.gii. The principle is to have the surface matches with your individual data.
- If you have cifti timeseries, it's probably easier to use -cifti-correlation and -cifti-gradient.

Let me know if anything is not clear.
Ting

**Comment by jcgriffis** [comment]
Thank you for the clarification; this is very helpful!

I do have one question, though:

"I didn't include the code for calculating the functional similarity here. The similarity of functional connectivity can be done using the matlab function 'paircorr_mod.m' or wb_command command '-cifti-correlation'. Note here, consistent with Wig2013, Gordon2014, the cortical feature is the similarity of functional connectivity, we can also use other cortical features as well (e.g. myelin, Glasser2016)."

So, would the functional similarity file here correspond directly to the 'corrofcorr_L.dconn.nii' that is saved by the Gordon et al., 'surface_parcellation.m' function?

Thanks again for your help,

Joseph

**Comment by jcgriffis** [comment]
Also, it looks like the watershed algorithm is applied outside of the x_gradient script to create the label maps?

**Comment by TingsterX** [comment]
Hi Joseph,

sorry, this message slipped my mind. I hope you have found the solution. 

Re: functional similarity. It is the similarity (Pearson'r) of the functional connectivity (Pearson'r), including the whole cortex data in the calculation. In the earlier test, I also checked the gradient of functional connectivity and the results are similar, though the previous study suggested that spatial and temporal similarities could be somewhat different (FIg1 Craddock et al., 2012 HBM). I think 'corrofcorr' is what you are looking for, but not 100% sure it includes the full cortex.

Re: watershed, in x_gradient script, it calls the matlab function to calculate all edge maps and all label maps. If you would like to create a single edge and label map, you'll need to average the edge maps to generate the edge probability map, then  run the watershed algorithm again to general one single edge/label map.

Best,
Ting
